firrtl/src/main/scala/firrtl/ir/IR.scala

```diff
@@ -329,6 +336,17 @@ object Print {
   }
 }
 
+case class ProbeDefine(info: Info, sink: Expression, probeExpr: Expression) extends Statement with UseSerializer
+
+case class ProbeForceInitial(info: Info, probe: Expression, value: Expression) extends Statement with UseSerializer
+case class ProbeReleaseInitial(info: Info, probe: Expression) extends Statement with UseSerializer
+case class ProbeForce(info: Info, clock: Expression, cond: Expression, probe: Expression, value: Expression)
```

Comment:
Is it useful/possible to be more narrow than "Expression" here for some checking? e.g., clock needs to be a "Clock" (unsure if that's a thing here that works in that way).

Reply:
I think we have to keep it as `Expression`, since people might pass in a `Reference` of unknown type, but which eventually evaluates down to `ClockType`. @jackkoenig does that sound correct?

Reply:
> I think we have to keep it as Expression, since people might pass in a Reference of unknown type, but which eventually evaluates down to ClockType. @jackkoenig does that sound correct?

This is true but immaterial--there is no more specific type in SFC IR to use here since the type of an Expression is not captured in the type system.

Reply:
To clarify @dtzSiFive this is just the internal "command" that Chisel creates when a user instantiates a probe. It is NOT the API that users call to instantiate the probe. That API is not present in this PR and will probably be in a followup PR, and will have additional checks etc like making sure the users pass a clock.